### PR TITLE
Create unit tests

### PR DESCRIPTION
**Pull request by antonlaukemper**
This would be the structure for all tests, change draft status as soon as all tests are implemented

### CODE REVIEW

**Review by rauwuckl — APPROVED** [review]
love it. all of it.

File: Backend/scala-backend/test/models/RelationshipModelSpec.scala [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    }
+    "create relations from a self-reported student to another self-reported student" in {
+      val rel1: RelationshipCC =
+        RelationshipCC(classId.get, srStudent1Id.get, srStudent2Id.get)
+
+      val success: Boolean =
+        awaitInf(relationshipModel.createRelationship(rel1))
+      success mustBe true
+    }
+    // "not create relations from not self-reported students to another student" in {
```
Comment (antonlaukemper) [inline Backend/scala-backend/test/models/RelationshipModelSpec.scala:84]: doe want these two tests to succeed? missing functionality?

File: Backend/scala-backend/test/models/SchoolClassModelSpec.scala [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+      createdClass3.className mustBe "test"
+    }
+    "return a schoolclassCC by id" in {
+      val returnedClass: SchoolClassCC =
+        awaitInf(classModel.getSchoolClass(createdClass1.get.id.get))
+      returnedClass.id.get mustBe 1
+      returnedClass.className mustBe "test"
+
+      // request nonexistent class
+      // do we want an exception here??
```
Comment (antonlaukemper) [inline Backend/scala-backend/test/models/SchoolClassModelSpec.scala:80]: question

File: Backend/scala-backend/test/models/SchoolClassModelSpec.scala [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+      notAuthenticatedTeacher2.isEmpty mustBe true
+    }
+    "return the status of a class by id" in {
+      // initial state is 0
+      val surveyStatus: Int =
+        awaitInf(classModel.getStatus(createdClass1.get.id.get))
+      surveyStatus mustBe 0
+
+      // nonexisting id
+      // again an Exception, do we want this?
```
Comment (antonlaukemper) [inline Backend/scala-backend/test/models/SchoolClassModelSpec.scala:135]: question

File: Backend/scala-backend/test/models/SchoolClassModelSpec.scala [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    "update the surveyStatus of a class if id is correct" in {
+      val surveyStatus: Int =
+        awaitInf(classModel.getStatus(createdClass1.get.id.get))
+      surveyStatus mustBe 0
+
+      val NumberOfChangedClasses: Int =
+        awaitInf(classModel.updateStatus(createdClass1.get.id.get, 1))
+      val surveyStatus1: Int =
+        awaitInf(classModel.getStatus(createdClass1.get.id.get))
+      surveyStatus1 mustBe 1
```
Comment (rauwuckl) [inline Backend/scala-backend/test/models/SchoolClassModelSpec.scala:150]: maybe replace with the enum

File: Backend/scala-backend/test/models/StudentModelSpec.scala [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+    }
+    "should return the studentID of old student if he is already self reported" in {
+       val student1: StudentCC =
+        StudentCC(None, "uniqueName", "encName", true, None)
+
+      val student1Id: Option[Int] =
+        awaitInf(studentModel.createStudent(student1, classId.get))
+      student1Id.get mustBe 1 // first student has id 1
+
+      // student with same name but this time the entry is self-reported
```
Comment (rauwuckl) [inline Backend/scala-backend/test/models/StudentModelSpec.scala:108]: delete comment

**Review by rauwuckl — APPROVED** [review]
great job

**Pull request merged**